panda/src/navigation/navMesh.cxx

```diff
@@ -0,0 +1,388 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file navMesh.cxx
+ * @author ashwini
+ * @date 2020-060-21
+ */
+
+
+#include "navMesh.h"
+#include "geom.h"
+#include "geomTrifans.h"
+#include <iostream>
+
+TypeHandle NavMesh::_type_handle;
+
+NavMesh::NavMesh():
+  _nav_mesh(0) {}
+
+NavMesh::~NavMesh() {
+  dtFreeNavMesh(_nav_mesh);
+  _nav_mesh = 0;
+}
+
+NavMesh::NavMesh(dtNavMesh *nav_mesh) {
+  _nav_mesh = nav_mesh;
+}
+
+NavMesh::NavMesh(NavMeshParams mesh_params):
+  _nav_mesh(0) {
+  memset(&(_params), 0, sizeof(_params));
```

Comment:
There's probably a more idiomatic way to do this in C++.  I guess `_params = {};` should do it.